irohad/ordering/impl/on_demand_ordering_service_impl.cpp

```diff
@@ -236,12 +236,16 @@ void OnDemandOrderingServiceImpl::packNextProposals(
       {round.block_round, currentRejectRoundConsumer(round.reject_round)});
 }
 
-void OnDemandOrderingServiceImpl::tryErase() {
-  while (round_queue_.size() > number_of_proposals_) {
-    auto &round = round_queue_.front();
-    proposal_map_.erase(round);
-    log_->info("tryErase: erased {}", round);
-    round_queue_.pop();
+void OnDemandOrderingServiceImpl::tryErase(
+    const consensus::Round &current_round) {
+  auto current_proposal = proposal_map_.find(current_round);
+  auto proposal_range_size = boost::size(
+      boost::make_iterator_range(proposal_map_.begin(), current_proposal));
+
+  while (proposal_range_size > number_of_proposals_
```

Comment:
quite a strange condition, because `proposal_range_size` is unchanged in the loop. maybe
```suggestion
  while (proposal_range_size-- > number_of_proposals_
```